src/pages/HomePage.js

```diff
@@ -1,21 +1,38 @@
-import React, { Component } from 'react';
+import React from 'react';
 
-import Banner from '../components/HomePage/Banner';
-import ButtonBar from '../components/HomePage/ButtonBar';
-import SleepyActionOutput from '../components/HomePage/SleepyActionOutput';
-import RandomDogOutput from '../components/HomePage/RandomDogOutput';
+import MainView from "../components/HomePage/MainView";
+import SearchResults from "../components/HomePage/SearchResultsArea/SearchResults";
+import ProductDescription from "../components/HomePage/ProductPlacementArea/ProductDescription";
 
-class HomePage extends Component {
-    render() {
-        return (
-            <React.Fragment>
-                <Banner />
-                <ButtonBar />
-                <SleepyActionOutput />
-                <RandomDogOutput />
-            </React.Fragment>
-        );
-    }
+function HomePage() {
+    
+    let [productDescriptionRef, setProductDescriptionRef] = React.useState(null);
+    let [searchResultsRef, setSearchResultsRef] = React.useState(null);
+
+    const scrollToProductDescription = () => {
+        productDescriptionRef.current.scrollIntoView({
+            behavior: "smooth",
+            block: "start"
+        });
+    };
+    const scrollToSearchResults = () => {
+        searchResultsRef.current.scrollIntoView({
+            behavior: "smooth",
+            block: "start"
+        });
+    };
+
+    return (
+        <React.Fragment>
+            <MainView 
+                scrollToProductDescription={scrollToProductDescription}
```

Comment:
These seem like they are being passed on for several levels. This is less than ideal, but I do not feel very well about putting them in the redux store either... :/

Would a solution using html `id`s work? It might be better code-wise

Reply:
As explained above, in React, the right way to interact with the html elements is via refs, instead of using plain JS

Reply:
Still, what about the redux store idea? Would be good to remove the multi-level prop passing - but is still kind of weird to use redux store for that...

Still, I'd be on board with that if you agreed, as I *reeeally* don't like passing props around so much

Reply:
I think it will stay like this, because:
1. It's only 2 levels of propagation
1. I don't think it makes much sense to use redux store for functionality related callbacks, as it should be used mostly for common data storage.
1. This is a really specific set of functions that does not relate with other parts of the application, even for future components, thus it should not be in the redux store

Reply:
> 1. It's only 2 levels of propagation

"only two" evolve into "only three" into "only four" and so on... It's best to tackle the problem right as it arises, and cut close to the root. I see your point though, two levels is not that bad, despite being less than ideal.

> 2. I don't think it makes much sense to use redux store for functionality related callbacks, as it should be used mostly for common data storage.

The redux store can really be used for whatever we want tbh. If this made sense then it sure would have its place there

> 3. This is a really specific set of functions that does not relate with other parts of the application, even for future components, thus it should not be in the redux store

On that we can agree. This makes sense.

I tried also investigating some options detailed [here](https://reactjs.org/docs/context.html), but I think it's probably overkill for this case, so I'd vote for keeping it as it is for now as well, and see how it evolves in the future (don't think this component will see much reuse anyway, as it's kinda specific).